### PR TITLE
[standalone_pem] Add examples and add option to fully static link.

### PR DESCRIPTION
**Pull request by etep**
Summary: We add some example scripts, a README.md file, and update the build file with an option to fully static link. Static linking may be desirable for deployment to non-containerized environments, but is an experimental feature since it statically links in glibc. We considered static linking with musl, but cannot figure out how to do that with C++.

Type of change: /kind feature

Test Plan: Existing tests. And, copied the standalone pem binary to a GKE node, then deleted all libs, then ran the binary. The binary ran (but could not find linux headers).

### CODE REVIEW

File: src/experimental/standalone_pem/BUILD.bazel [hunk 1]
```diff
@@ -57,6 +57,7 @@ pl_cc_library(
 pl_cc_binary(
```
Comment (ddelnano) [inline src/experimental/standalone_pem/BUILD.bazel:57]: Did you consider linking a different libc like musl? I believe static linking glibc is generally [discouraged](https://stackoverflow.com/questions/57476533/why-is-statically-linking-glibc-discouraged), although I don't have experience with these problems.

**Review by JamesMBartlett — CHANGES_REQUESTED** [review]
Fully statically linking GCC will cause unexpected errors, especially with network calls. If we want to do fully statically linked binaries we have to use musl.

**Comment by etep** [comment]
Agreed, needs to use `musl`. How much scope is there in switching to `musl` ... is that something like a new `cc_toolchain`?

**Comment by etep** [comment]
It is not ok to statically link to glibc. Planning to setup a tool chain to use `musl` for this purpose.

**Pull request closed without merging**